test/unit/regexes.test.ts

```diff
@@ -20,7 +22,7 @@ describe('Unit: Regexes', () => {
       '/dir.dir/file.f.txt',
       '/dir space dir/dir dir/file-file.txt'
     ]
-    const invalidPaths = ['/', 'file', 'dir/file.txt', '/ file', '/file ']
+    const invalidPaths = ['/', ' /', '/ ', '/ file', '/file ']
```

Comment:
I know this is out of scope for this PR, but these are actually valid Unix paths. Maybe not on Windows, but Unix is surprisingly loose.

Reply:
These constraints are specific to the sync server. Perhaps they're unnecessary constraints and we could be clearer on the path requirements and improve upon the spec.

Reply:
The good news is that we can loosen them later, and it's a non-breaking change. I think we can live with that.